homeassistant/components/matter/adapter.py

```diff
@@ -52,13 +52,62 @@ def register_platform_handler(
 
     async def setup_nodes(self) -> None:
         """Set up all existing nodes and subscribe to new nodes."""
-        for node in await self.matter_client.get_nodes():
+        for node in self.matter_client.get_nodes():
             self._setup_node(node)
 
         def node_added_callback(event: EventType, node: MatterNode) -> None:
             """Handle node added event."""
             self._setup_node(node)
 
+        def endpoint_added_callback(event: EventType, data: dict[str, int]) -> None:
+            """Handle endpoint added event."""
+            node = self.matter_client.get_node(data["node_id"])
+            self._setup_endpoint(node.endpoints[data["endpoint_id"]])
+
+        def endpoint_removed_callback(event: EventType, data: dict[str, int]) -> None:
+            """Handle endpoint removed event."""
+            server_info = cast(ServerInfoMessage, self.matter_client.server_info)
+            try:
+                node = self.matter_client.get_node(data["node_id"])
+            except KeyError:
```

Comment:
The library shouldn't leak a KeyError. The library should handle this and raise a library specific exception if needed.